lib/open_api_spex/cast/object.ex

```diff
@@ -112,6 +114,31 @@ defmodule OpenApiSpex.Cast.Object do
     end)
   end
 
+  # Pass additional properties through when `additionalProperties` is true.
+  # Map string keys are not converted to atoms. That would require calling `String.to_atom/1`, which is not safe.
+  defp cast_additional_properties(%{schema: %{additionalProperties: true}} = ctx, original_value) do
```

Comment:
When `additionalProperties` is `true` or `nil` This is the right behaviour.

For completeness we should also support validating additional properties against a schema - can be a separate PR.

I’d like to add a Protocol to give users control over how additional properties are handled, something like the Extendable protocol - also a separate PR.